clients/js/src/utils/operations.ts

```diff
@@ -1,9 +1,33 @@
-const compareSemanticIsGreaterImp = function (version_a: string, version_b: string) {
-    return version_a.localeCompare(version_b, undefined, {numeric:true, sensitivity:'base'}) > 0;
+const compareSemantic = function (version_a: string, version_b: string): number {
+    return version_a.localeCompare(version_b, undefined, { numeric: true, sensitivity: 'base' });
```

Comment:
the `locales` parameter and `options` parameter [are not supported](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare#browser_compatibility) on android webview
<img width="833" alt="Screenshot 2024-05-24 at 12 21 32 PM" src="https://github.com/juspay/superposition/assets/15166178/d9f008a6-0bdd-4ff7-9aa2-2f31e3fb9e49">